plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -126,4 +239,99 @@ AS SELECT sum(value) AS s FROM iceberg2.tpch.common_base_table
         assertUpdate(defaultIceberg, "DROP TABLE common_base_table");
         assertUpdate("DROP MATERIALIZED VIEW mv_on_iceberg2");
     }
+
+    private int validateResultContains2AndReturnSecondValue(List<MaterializedRow> materializedRows)
+    {
+        assertThat(materializedRows.size()).isEqualTo(2);
+        assertThat(materializedRows.stream().filter(value -> (int) value.getField(0) == 2).count()).isEqualTo(1);
+        return materializedRows.stream().mapToInt(value -> (int) value.getField(0)).filter(value -> value != 2).findFirst().orElseThrow();
+    }
+
+    public static class RandomTableFunction
+            extends AbstractConnectorTableFunction
+    {
+        public RandomTableFunction()
+        {
+            super("system", "random_function", List.of(), GENERIC_TABLE);
+        }
+
+        @Override
+        public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments, ConnectorAccessControl accessControl)
+        {
+            return TableFunctionAnalysis.builder()
+                    .handle(new RandomTableFunctionHandle())
+                    .returnedType(new Descriptor(ImmutableList.of(new Descriptor.Field("random_value", Optional.of(INTEGER)))))
+                    .build();
+        }
+    }
+
+    public  static class RandomTableFunctionHandle
```

Comment:
can you please fix checkstyle so that it bails out on double spaces after `public`?

Reply:
I will try in a separate PR